app/controllers/downvotes_controller.rb

```diff
@@ -0,0 +1,19 @@
+class DownvotesController < ApplicationController
+  before_action :require_login, only: [:create]
+
+  def create
+    @vote = parent.votes.for_user(current_user).first_or_initialize
+    @vote.score = -1
+    @vote.save
```

Comment:
good to know. I thought that was `update_column`, but it seems `update_attribute` skips validation but invokes callbacks. y u do dis, rails?